src/main/fc/rc.c

```diff
@@ -416,19 +442,18 @@ static FAST_CODE void processRcSmoothingFilter(void)
             const bool ready = (currentTimeMs > 1000) && (targetPidLooptime > 0);
             if (ready) { // skip during FC initialization
                 // Wait 1000ms after power to let the PID loop stabilize before starting average frame rate calculation
-                if (rxIsReceivingSignal() && isRxIntervalValid) {
+                if (rxIsReceivingSignal() && isRxRateValid) {
 
-                    static uint16_t previousRxIntervalUs;
                     if (abs(currentRxIntervalUs - previousRxIntervalUs) < (previousRxIntervalUs - (previousRxIntervalUs / 8))) {
                         // exclude large steps, eg after dropouts or telemetry
                         // by using interval here, we catch a dropout/telemetry where the inteval increases by 100%, but accept
                         // the return to normal value, which is only 50% different from the 100% interval of a single drop, and 66% of a return after a double drop.
-                        static float prevRxRateHz;
+                        static float prevSmoothedRxRateHz;
                         // smooth the current Rx link frequency estimates
                         const float kF = 0.1f; // first order lowpass smoothing filter coefficient
-                        const float smoothedRxRateHz = prevRxRateHz + kF * (currentRxRateHz - prevRxRateHz);
-                        prevRxRateHz = smoothedRxRateHz;
-      
+                        const float smoothedRxRateHz = prevSmoothedRxRateHz + kF * (currentRxRateHz - prevSmoothedRxRateHz);
```

Comment:
(this would be much cleaner as PT1 filter)

Reply:
True, but I am hopeless at coding those filters, and this is simple :-) 
On the plus side, it just adds 1/10th of the change in rate each time a new packet arrives, and in a sense is self-explanatory.  It doesn't have a cutoff frequency as such.
For now I will leave it, I modified the comment slightly.
Mostly we just get single packet dropouts, and this means that a single dropout will cause only a 10% deviation in the estimate.  It takes a string of them, or some really big ones, to make a big difference.

Reply:
Its just minor thing..
it is mathematically equivalent to digital RC (pt1 in bf) filter. So it is IMO better to use filter code instead of 'inlining' it.
It is so easier to reason about filter behavior - it is standard concept, there is a lot of literature available. You can easily calculate cutoff or time constant and answer questions like 'how long it takes for output to settle'.
Also, if necessary, it is easier to replace with higher order filter (likely not necessary in this case)